### PR TITLE
gh #59 Added ALLM Set/Get APIs

### PR DESCRIPTION
**Pull request by SrigayathryPugazhenthi**

### CODE REVIEW

File: include/dsVideoPort.h [hunk 1]
```diff
@@ -660,6 +660,58 @@ dsError_t dsSetForceDisable4KSupport(intptr_t handle, bool disable);
...
+ *                         ( @a true to enable, @a false to disable)
+ *
+ * @return dsError_t                      -  Status
+ * @retval dsERR_NONE                     -  Success
+ * @retval dsERR_NOT_INITIALIZED          -  Module is not initialised
+ * @retval dsERR_INVALID_PARAM            -  Parameter passed to this function is invalid
+ * @retval dsERR_OPERATION_NOT_SUPPORTED  -  The attempted operation is not supported
+ * @retval dsERR_GENERAL                  -  Underlying undefined platform error
+ *
+ * @pre dsVideoPortInit() and dsGetVideoPort() must be called before calling this API.
```
Comment (apatel859) [inline include/dsVideoPort.h:680]: why to call dsGetVideoPort as pre for set call?

File: include/dsVideoPort.h [hunk 2]
```diff
@@ -660,6 +660,58 @@ dsError_t dsSetForceDisable4KSupport(intptr_t handle, bool disable);
...
+ *                          ( @a true when ALLM mode is enabled or @a false otherwise)
+ *
+ * @return dsError_t                      -  Status
+ * @retval dsERR_NONE                     -  Success
+ * @retval dsERR_NOT_INITIALIZED          -  Module is not initialised
+ * @retval dsERR_INVALID_PARAM            -  Parameter passed to this function is invalid
+ * @retval dsERR_OPERATION_NOT_SUPPORTED  -  The attempted operation is not supported
+ * @retval dsERR_GENERAL                  -  Underlying undefined platform error
+ *
+ * @pre dsVideoPortInit() and dsGetVideoPort() must be called before calling this API.
```
Comment (apatel859) [inline include/dsVideoPort.h:707]: remove dsGetVideoPort

File: include/dsVideoPort.h (hunk 1 above)
Comment (Srigayathry) [reply include/dsVideoPort.h:680]: To get the video port handle

File: include/dsVideoPort.h (hunk 2 above)
Comment (Srigayathry) [reply include/dsVideoPort.h:707]: Is it not needed to mention dsGetVideoPort call to get the port handle

**Review by apatel859 — APPROVED** [review]

**Comment by hari22yuva** [comment]
> _ No description provided. _

Please add description

File: include/dsVideoPort.h [hunk 3]
```diff
@@ -660,6 +660,58 @@ dsError_t dsSetForceDisable4KSupport(intptr_t handle, bool disable);
  */
 dsError_t dsGetForceDisable4KSupport(intptr_t handle, bool *disable);
 
+/**
+ * @brief Enables/Disables ALLM mode for HDMI output video port.
```
Comment (hari22yuva) [inline include/dsVideoPort.h:664]: how do video port know ALLM mode is supported?
Need test cases of these API's

File: include/dsVideoPort.h [hunk 4]
```diff
@@ -660,6 +660,58 @@ dsError_t dsSetForceDisable4KSupport(intptr_t handle, bool disable);
...
+
+/**
+ * @brief Checks whether ALLM mode of HDMI output video port is enabled or not.
+ *
+ * For Source devices, this function indicates whether ALLM mode for specified HDMI output video port is enabled or not.
+ * By default ALLM mode is disabled on bootup.
+ * For sink devices, this function returns dsERR_OPERATION_NOT_SUPPORTED always.
+ *
+ * @param[in]  handle   - Handle of the video port returned from dsGetVideoPort()
+ * @param[out] enabled  - Flag to hold the enabled status of ALLM mode for given HDMI output video port.
```
Comment (srinivasgtl) [inline include/dsVideoPort.h:697]: pointer to hold the ALLM Mode enable/disable status.

**Review by hari22yuva — COMMENTED** [review]
Please address the review comments

File: include/dsVideoPort.h (hunk 3 above)
Comment (Srigayathry) [reply include/dsVideoPort.h:664]: Its SOC feature to enable low latency for the hdmi out port. will check if any capability API needed

File: include/dsVideoPort.h [hunk 5]
```diff
@@ -660,6 +660,58 @@ dsError_t dsSetForceDisable4KSupport(intptr_t handle, bool disable);
  */
 dsError_t dsGetForceDisable4KSupport(intptr_t handle, bool *disable);
 
+/**
+ * @brief Enables/Disables ALLM mode for HDMI output video port.
+ *
+ * For source devices, this function enables or disables the ALLM mode for specified HDMI output video port.
```
Comment (srinivasgtl) [inline include/dsVideoPort.h:666]: Continuing the first statement.  Also, mention "Enabling or disabling should follow the HDMI 2.1 Specification.  ALLM bit on HF-VSDB should be considered before setting ALLM_Mode bit on the HF-VSIF"

File: include/dsVideoPort.h (hunk 3 above)
Comment (SrigayathryPugazhenthi) [reply include/dsVideoPort.h:664]: Its SOC feature to enable low latency for the hdmi out port. will check if any capability API needed

**Pull request closed without merging**